service/collector_test.go

```diff
@@ -373,6 +373,29 @@ func TestCollectorShutdownBeforeRun(t *testing.T) {
 	assert.Equal(t, Closed, col.GetState())
 }
 
+func TestCollectorShutdownBeforeRun(t *testing.T) {
+	factories, err := componenttest.NopFactories()
+	require.NoError(t, err)
+
+	cfgProvider, err := NewConfigProvider(newDefaultConfigProviderSettings([]string{filepath.Join("testdata", "otelcol-nop.yaml")}))
+	require.NoError(t, err)
+
+	set := CollectorSettings{
+		BuildInfo:      component.NewDefaultBuildInfo(),
+		Factories:      factories,
+		ConfigProvider: cfgProvider,
+		telemetry:      newColTelemetry(featuregate.NewRegistry()),
+	}
+	col, err := New(set)
```

Comment:
There's something odd -- have you swapped the test names by any chance?

Reply:
I use the split view which should prevent this but I checked out the actual files in any case: the older version has `require.NotPanics(t, func() { col.Shutdown() })` in `TestCollectorShutdownBeforeRun`, which the new doesn't. This line does show up in `TestCollectorWithNoMetrics` though.

Reply:
You are right, thanks for catching that! Fixed